flowetl/etl/etl/model.py

```diff
@@ -0,0 +1,87 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+# -*- coding: utf-8 -*-
+"""
+Define a DB model for storing the process of ingestion
+"""
+
+import pendulum
+from pendulum.date import Date as pendulumDate
+
+from sqlalchemy import Column, String, DateTime, Date, Integer
+from sqlalchemy.ext.declarative import declarative_base
+from sqlalchemy.orm.session import Session
+
+Base = declarative_base()
+
+# pylint: disable=too-few-public-methods
+class ETLRecord(Base):
+    """
+    DB Model for storing the process of ingestion
+    for found files.
+    """
+
+    __tablename__ = "etl_records"
+    __table_args__ = {"schema": "etl"}
+
+    id = Column(Integer, primary_key=True)
+    file_name = Column(String)
+    cdr_type = Column(String)
+    cdr_date = Column(Date)
+    state = Column(String)
+    timestamp = Column(DateTime)
+
+    def __init__(
+        self, *, file_name: str, cdr_type: str, cdr_date: pendulumDate, state: str
+    ):
+
+        # This should be set more globally - not using enums
+        # because don't have a migration strategy in place
```

Comment:
Out of interest, what do you mean here? Why does a missing migration strategy prevent using enums?

Reply:
Because you have no way of keeping the data in this table and adding a new type very easily...